test/parallel/test-error-codes-impl-doc.js

```diff
@@ -0,0 +1,156 @@
+const results = [];
+function check(ok, type, name, reason) {
+  const label = `${type}: ${name} - ${reason}`;
+  if (report) {
+    if (!ok) results.push(label);
+  } else {
+    assert.ok(ok, label);
+  }
+}
+
+const set = {
+  all: new Set(),
+  js: new Set(), // found in natives
+  doc: new Set(), // mentioned in doc/api/*.md
+  cpp: new Set(), // found in src/node_errors.h
+  noargs: new Set(), // subset of `js` which is constructed without arguments
+  // `set.manual` contains errors manually created from js and not defined in
+  // internals/errors. That is not scanned now, update this list if required.
+  manual: new Set(['ERR_HTTP2_ERROR', 'ERR_UNKNOWN_BUILTIN_MODULE']),
+  // `set.examples` contains errors mentioned in documentation purely for
+  // demonstration purposes. These errors are not required to be present.
+  examples: new Set(['ERR_ERROR_1'])
+};
+
+const root = path.join(__dirname, '..', '..');
+
+// File containing error definitions
+const jsdata = natives['internal/errors'];
+// File containing error documentation
+const docdata = fs.readFileSync(path.join(root, 'doc/api/errors.md'), 'utf-8');
+// File containing cpp-side errors
+const cppdata = fs.readFileSync(path.join(root, 'src/node_errors.h'), 'utf-8');
+// Directory containing documentation
+const docdir = path.join(root, 'doc/api/');
+
+function addSource(source, type) {
+  // eslint-disable-next-line node-core/no-unescaped-regexp-dot
+  const re = /(.)?(ERR_[A-Z0-9_]+)(..)?/g;
+  let match;
+  while (match = re.exec(source)) {
+    if (match[1] === '_') continue; // does not start with ERR_
+    const name = match[2];
+    if (type === 'doc' && set.examples.has(name)) continue; // is an example
+    set.all.add(name);
+    set[type].add(name);
+    if (type === 'js' && match[3] === '()') {
+      // Is directly called without arguments from js, we will try that
+      set.noargs.add(name);
+    }
+  }
+}
+
+// Add all errors from JS natives
+for (const file of Object.keys(natives)) addSource(natives[file], 'js');
+// Add all errors from src/node_errors.h
+addSource(cppdata, 'cpp');
+// Add all errors from doc/api/*.md files
+for (const file of fs.readdirSync(docdir)) {
+  if (!file.endsWith('.md')) continue;
+  const data = fs.readFileSync(path.join(docdir, file), 'utf-8');
+  addSource(data, 'doc');
+}
+
+// Check that we have all js errors
+for (const name of set.js) {
+  if (set.manual.has(name)) continue;
+  const defined = jsdata.includes(`E('${name}',`);
+  check(defined, 'js', name, 'missing JS implementation (source)');
+  if (defined) {
+    check(errors[name], 'js', name, 'missing JS implementation (runtime)');
+  }
+}
+
+// Check that we have can initialize errors called without args
+for (const name of set.noargs) {
+  if (!errors[name]) continue; // Already catched that above
+  let ok = true;
+  try {
+    new errors[name]();
+  } catch (e) {
+    ok = false;
+  }
+  check(ok, 'init', name, 'failed 0 args init, but is called with "()"');
```

Comment:
This comment can be a bit confusing, as `0` can be read as an argument. Maybe `failed no args init`?

Reply:
Will change to `failed init without args`.